fcrepo-http-commons/src/test/java/org/fcrepo/http/commons/api/rdf/HttpIdentifierConverterTest.java

```diff
@@ -0,0 +1,203 @@
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(baseUrl, httpUri);
+    }
+
+    @Test
+    public void testFirstLevelWithMemento() {
+        final String memento = "20190926133245";
+        final String baseUid = getUniqueId();
+        final String baseUrl = uriBase + "/" + baseUid;
+        final String testUri = baseUrl + "/" + FCR_VERSIONS + memento;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(baseUrl, httpUri);
+    }
+
+    @Test
+    public void testSecondLevel() {
+        final String baseUid = getUniqueId() + "/" + getUniqueId();
+        final String testUri = uriBase + "/" + baseUid;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(testUri, httpUri);
+    }
+
+    @Test
+    public void testSecondLevelWithAcl() {
+        final String baseUid = getUniqueId() + "/" + getUniqueId();
+        final String baseUrl = uriBase + "/" + baseUid;
+        final String testUri = baseUrl + "/" + FCR_ACL;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(baseUrl, httpUri);
+    }
+
+    @Test
+    public void testSecondLevelWithMetadata() {
+        final String baseUid = getUniqueId() + "/" + getUniqueId();
+        final String baseUrl = uriBase + "/" + baseUid;
+        final String testUri = baseUrl + "/" + FCR_METADATA;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(baseUrl, httpUri);
+    }
+
+    @Test
+    public void testSecondLevelWithVersions() {
+        final String baseUid = getUniqueId() + "/" + getUniqueId();
+        final String baseUrl = uriBase + "/" + baseUid;
+        final String testUri = baseUrl + "/" + FCR_VERSIONS;
+        final String fedoraId = converter.convert(testUri);
+        assertEquals("info:fedora/" + baseUid, fedoraId);
+        final String httpUri = converter.reverse().convert(fedoraId);
+        assertEquals(baseUrl, httpUri);
+    }
+
+    @Test
+    public void testSecondLevelWithMemento() {
+        final String memento = "20190926133245";
+        final String baseUid = getUniqueId() + "/" + getUniqueId();
+        final String baseUrl = uriBase + "/" + baseUid;
+        final String testUri = baseUrl + "/" + FCR_VERSIONS + memento;
```

Comment:
slash again?

Reply:
Also, seems like the implication is that a memento shares the same identifier as the original. I think that's okay, it would just be represented as an instance of the same resource but with a memento date time.

Reply:
Or should I just strip the base Url off and prepend `info:fedora/`?
So

| External | Internal |
| --- | --- |
| http://server:8080/fcrepo/rest/bob | info:fedora/bob |
| http://server:8080/fcrepo/rest/bob/fcr:acl | info:fedora/bob/fcr:acl |
http://server:8080/fcrepo/rest/bob/fcr:versions | info:fedora/bob/fcr:versions |

etc and let the persistence layer do the rest of the translating?